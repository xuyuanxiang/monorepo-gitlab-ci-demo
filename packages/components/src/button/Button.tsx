import React from 'react';
import clsx from 'clsx';
import styled from 'styled-components';
import { IButtonProps } from './prop-types';

const StyledButton = styled.button`
  &.secondary {
    border: 1px solid #d9af5c;
    color: #d9af5c;
    &:active {
      background: #fbf7ee;
    }
  }
  &.primary {
    border: 0;
    background: #d9af5c;
    color: #ffffff;
    &:active {
      background: #cd962e;
    }
  }
  &.negative {
    border: 1px solid #d23f31;
    color: #d23f31;
    &:active {
      background: #fff3f2;
    }
  }
  &:disabled {
    background: #cccccc;
    color: #ffffff;
    border: 0;
  }
  &.standard {
    display: block;
    width: 100%;
  }
  &.small {
    font-size: 13px;
    height: 30px;
    line-height: 30px;
    padding: 0 8.5px;
  }
  max-width: 100%;
  padding: 0 16px;
  border-radius: 4px;
  outline: none;
  font-size: 18px;
  height: 48px;
  line-height: 48px;
  font-weight: 500;
  appearance: none;
  box-sizing: border-box;
  margin: 0;
  overflow: hidden;
  text-overflow: ellipsis;
  word-break: break-word;
  white-space: nowrap;
  background: #ffffff;
  user-select: none;
  touch-action: none;
  vertical-align: text-top;
`;

export function Button({
  children,
  scene = 'primary',
  spec = 'standard',
  className,
  ...buttonProps
}: IButtonProps): JSX.Element {
  return (
    <StyledButton className={clsx(scene, spec, className)} {...buttonProps}>
      {children}
    </StyledButton>
  );
}
