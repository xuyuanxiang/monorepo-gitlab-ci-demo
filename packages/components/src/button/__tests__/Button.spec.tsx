import React from 'react';
import 'jest-styled-components';
import { mount, render } from 'enzyme';
import { Button } from '../Button';

describe('<Button/>', () => {
  it('should be clicked', () => {
    const handleClick = jest.fn();
    const ele = mount(<Button onClick={handleClick}>Click me!</Button>);
    ele.simulate('click');
    expect(handleClick).toBeCalledTimes(1);
    expect(ele.text()).toBe('Click me!');
  });
  it('should render default props', () => {
    const ele = render(<Button />);
    expect(ele).toMatchSnapshot();
  });
});
