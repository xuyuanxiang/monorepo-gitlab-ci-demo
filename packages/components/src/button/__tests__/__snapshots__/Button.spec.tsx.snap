// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render default props 1`] = `
.c0 {
  max-width: 100%;
  padding: 0 16px;
  border-radius: 4px;
  outline: none;
  font-size: 18px;
  height: 48px;
  line-height: 48px;
  font-weight: 500;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  margin: 0;
  overflow: hidden;
  text-overflow: ellipsis;
  word-break: break-word;
  white-space: nowrap;
  background: #ffffff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: none;
  vertical-align: text-top;
}

.c0.secondary {
  border: 1px solid #d9af5c;
  color: #d9af5c;
}

.c0.secondary:active {
  background: #fbf7ee;
}

.c0.primary {
  border: 0;
  background: #d9af5c;
  color: #ffffff;
}

.c0.primary:active {
  background: #cd962e;
}

.c0.negative {
  border: 1px solid #d23f31;
  color: #d23f31;
}

.c0.negative:active {
  background: #fff3f2;
}

.c0:disabled {
  background: #cccccc;
  color: #ffffff;
  border: 0;
}

.c0.standard {
  display: block;
  width: 100%;
}

.c0.small {
  font-size: 13px;
  height: 30px;
  line-height: 30px;
  padding: 0 8.5px;
}

<button
  class="c0 primary standard"
/>
`;
